spec/rubocop/cop/style/frozen_string_literal_comment_spec.rb

```diff
@@ -249,6 +249,39 @@
       RUBY
     end
 
+    it 'accepts a frozen string literal comment after other comments' do
+      expect_no_offenses(<<~RUBY)
+        #!/usr/bin/env ruby
+        # encoding: utf-8
+        #
+        # These are more comments
+        # frozen_string_literal: false
+        puts 1
+      RUBY
+    end
+
+    it 'registers an offense for having a frozen string literal comment ' \
+       'under ruby code' do
+      expect_offense(<<~RUBY)
+        # encoding: utf-8
+        ^ Missing frozen string literal comment.
+
+        puts 1
+        # frozen_string_literal: true
+      RUBY
+
+      # Since we're only looking at the leading comments, a
+      # `frozen_string_literal` comment elsewhere in the code is invisible
+      # to this cop so the autocorrect won't remove it.
```

Comment:
I would be happy to take a stab at implementing this, but I'm hesitant since the `autocorrect` method takes a specific node and we'd have to update it to "correct" code elsewhere in the source outside of that node.

Reply:
I think this case does not need to be taken care of. Because a frozen string magic comment written other than the reading comments may be an intended content in the source code comments.